### PR TITLE
[VENTUS][fix] Add ctz function implementation

### PR DESCRIPTION
**Pull request by zhoujingya**

### CODE REVIEW

File: libclc/generic/lib/SOURCES [hunk 1]
```diff
@@ -67,6 +67,7 @@ integer/abs.cl
 integer/abs_diff.cl
 integer/add_sat.cl
 integer/clz.cl
+integer/ctz.cl
```
[the remark this answers is not in the document]
Comment (zhoujingya) [reply libclc/generic/lib/SOURCES:70]: > 1. ctz is similar to clz, why not use the "__builtin_ctz" method?
> 2. Learn from clz：
> 
> > [e78344d](https://github.com/THU-DSP-LAB/llvm-project/commit/e78344dfae4abb858bb04bc1fcafb57f4bd9f49d)
> > [943057a](https://github.com/THU-DSP-LAB/llvm-project/commit/943057a2885a0d886de142a2a9beb227ca78ba23)

you are right, we now change to `builtin` function

**Review by Jules-Kong — COMMENTED** [review]
Don't need a file like "clz.inc"?

**Comment by zhoujingya** [comment]
> Reference in new issue

No need

**Comment by zhoujingya** [comment]
> Don't need a file like "clz.inc"?



> Don't need a file like "clz.inc"?

No need

File: libclc/generic/lib/integer/ctz.cl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+_CLC_OVERLOAD _CLC_DEF uchar ctz(uchar x) {
+  return x ? ctz((ushort)x) : 8;
+}
+
+_CLC_OVERLOAD _CLC_DEF int ctz(int x) {
+  return x ? __builtin_ctz(x) : 32;
+}
+
+_CLC_OVERLOAD _CLC_DEF uint ctz(uint ux) {
+  return ux ? __builtin_ctz(ux) : 32;
```
Comment (Jules-Kong) [inline libclc/generic/lib/integer/ctz.cl:24]: Consistent with llvm way. It is recommended to change "ux" to "x", like "uchar" uses "x".

File: libclc/generic/lib/integer/ctz.cl [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+_CLC_OVERLOAD _CLC_DEF long ctz(long x) {
+  return x ? __builtin_ctzl(x) : 64;
+}
+
+_CLC_OVERLOAD _CLC_DEF ulong ctz(ulong ux) {
+  return ux ? __builtin_ctzl(ux) : 64;
```
Comment (Jules-Kong) [inline libclc/generic/lib/integer/ctz.cl:32]: as above.

File: libclc/generic/lib/integer/ctz.cl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+_CLC_OVERLOAD _CLC_DEF short ctz(short x) {
+  return x ? __builtin_ctzs(x) : 16;
+}
+
+_CLC_OVERLOAD _CLC_DEF ushort ctz(ushort x) {
+  return x ? __builtin_ctzs(x) : 16;
+}
+
+_CLC_OVERLOAD _CLC_DEF char ctz(char x) {
```
Comment (Jules-Kong) [inline libclc/generic/lib/integer/ctz.cl:12]: Consistent with llvm way. It is recommended to adjust the situation of char and uchar in the front.

**Comment by Jules-Kong** [comment]
Can you provide a minimal use case and how to reproduce it?

**Comment by zhoujingfighting** [comment]
> Can you provide a minimal use case and how to reproduce it?

You can run CTS conformance test `build/test_conformance/integer_ops`, 'integer_ctz' test case

File: libclc/generic/lib/integer/ctz.cl [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#include "../clcmacro.h"
```
Comment (Jules-Kong) [inline libclc/generic/lib/integer/ctz.cl:1]: Supplement the vector types supported by the function through the gentype.inc file.

**Comment by Jules-Kong** [comment]
> > Can you provide a minimal use case and how to reproduce it?
> 
> You can run CTS conformance test `build/test_conformance/integer_ops`, 'integer_ctz' test case

@wangqinfan and I have agreed to provide these to reduce unnecessary processes. Will you do it for him?

**Comment by Jules-Kong** [comment]
Please merge the commits into one.

**Review by Jules-Kong — APPROVED** [review]
LGTM

**Pull request merged**